VirtualSurfaces/WinComp.cpp

```diff
@@ -129,6 +120,48 @@ void WinComp::PrepareVisuals(HANDLE handle)
 	redirectVisual.Children().InsertAtTop(this->redirectVisual);
 
 	visualTarget->SetRoot(reinterpret_cast<ABI::Windows::UI::Composition::IVisual*>(winrt::get_abi(redirectVisual)));
+
+	Windows::System::Threading::WorkItemHandler workItemHandler([this, redirectVisual](Windows::Foundation::IAsyncAction const&)
+	{
+		CoreIndependentInputSource source{ nullptr };
+
+		if (Windows::Foundation::Metadata::ApiInformation::IsTypePresent(winrt::name_of<CoreIndependentInputSourceController>()))
+		{
+			CoreIndependentInputSourceController controller = CoreIndependentInputSourceController::CreateForVisual(redirectVisual);

```

Comment:
`controller` goes out of scope and prevents touch input from working in the “primary” window. Making it a field seems to restore functionality

Reply:
fixed